### PR TITLE
adds a test that runs the db seed, making sure it's compatible with latest schema

### LINKED ISSUE #387: Test for db: seed
- [ ] Check for closure of PR #386 / merging
- [ ] Add a test for db: seed command 

 [Context](https://github.com/sfbrigade/compass/pull/386) : "One other thought for future improvement: Should there be a test to make sure the db:seed command doesn't fail in the future if there are schema changes to the database?"

### PR DESCRIPTION
**Pull request by thomhickey**
- creates a test db
- runs 'seedfile' from src/backend/db/lib/seed.ts
- queries the db for both students and users
- asserts that there are >0 students and users
- closes #387

### CODE REVIEW

**Comment by katconnors** [comment]
Hey @thomhickey, assuming that this only tests the seedfile, I think it would be good to decouple issue #387 and make a separate issue for this. 

The script that I created in #387 is different.

File: src/backend/db/lib/seed.test.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import test from "ava";
+import { getTestDatabase } from "@/backend/tests/fixtures/get-test-database";
+import { getDb } from "@/backend/db/lib/get-db";
+import { seedfile } from "@/backend/db/lib/seed";
+
+test("seedFile works after schema changes", async (t) => {
```
Comment (mrabbitt) [inline src/backend/db/lib/seed.test.ts:6]: Minor suggestion: I think "seedFile works with current schema" would be a slightly better title. The "after schema changes" part could imply that the test itself is applying schema changes before running seedFile.

**Review by mrabbitt — COMMENTED** [review]
Looks good to me except for a minor comment on test naming.

I tested that this actually fails when backwards incompatible schema changes are made by replacing all instances of `last_name` with `surname` in `src/backend/db/migrations/1_initial-migrations.sql` and then running the test. It failed exactly as I'd expect with the error message below. ✅ 

Thank you, @thomhickey !

```
  Uncaught exception in src/backend/db/lib/seed.test.ts

  error: terminating connection due to administrator command

  › Parser.parseErrorMessage (node_modules/pg-protocol/dist/parser.js:287:98)
  › Parser.handlePacket (node_modules/pg-protocol/dist/parser.js:126:29)
  › Parser.parse (node_modules/pg-protocol/dist/parser.js:39:38)
  › Socket.<anonymous> (node_modules/pg-protocol/dist/index.js:11:42)

  ✘ [fail]: seedFile works after schema changes Rejected promise returned by test
  ✘ src/backend/db/lib/seed.test.ts exited with a non-zero exit code: 1
  ─

  seedFile works after schema changes

  Rejected promise returned by test. Reason:

  Error {
    message: 'column "last_name" of relation "user" does not exist',
  }

  › Parser.parseErrorMessage (node_modules/pg-protocol/dist/parser.js:287:98)
  › Parser.handlePacket (node_modules/pg-protocol/dist/parser.js:126:29)
  › Parser.parse (node_modules/pg-protocol/dist/parser.js:39:38)
  › Socket.<anonymous> (node_modules/pg-protocol/dist/index.js:11:42)
  › PostgresConnection.executeQuery (node_modules/kysely/dist/cjs/dialect/postgres/postgres-driver.js:91:69)
  › <anonymous> (node_modules/kysely/dist/cjs/query-executor/query-executor-base.js:37:28)
  › DefaultConnectionProvider.provideConnection (node_modules/kysely/dist/cjs/driver/default-connection-provider.js:12:20)
  › DefaultQueryExecutor.executeQuery (node_modules/kysely/dist/cjs/query-executor/query-executor-base.js:36:16)
  › InsertQueryBuilder.execute (node_modules/kysely/dist/cjs/query-builder/insert-query-builder.js:517:24)
  › InsertQueryBuilder.executeTakeFirst (node_modules/kysely/dist/cjs/query-builder/insert-query-builder.js:532:26)
  › InsertQueryBuilder.executeTakeFirstOrThrow (node_modules/kysely/dist/cjs/query-builder/insert-query-builder.js:544:24)
  › seed (src/backend/tests/seed.ts:6:17)
  › Object.beforeTemplateIsBaked (src/backend/tests/fixtures/get-test-database.ts:21:24)

  ─

  1 test failed
  1 uncaught exception
```

**Review by mrabbitt — APPROVED** [review]

File: src/backend/db/lib/seed.test.ts (hunk 1 above)
Comment (thomhickey) [reply src/backend/db/lib/seed.test.ts]: great callout @mrabbitt ! ... changed in 2c0415ac4d099d6c4602222a2dbd7ac660c91d9e to 'seedfile works with current schema'

**Comment by thomhickey** [comment]
thanks for the negative test to make sure it fails @mrabbitt .. awesome

**Review by mrabbitt — APPROVED** [review]
LGTM ✅

**Review by KCCPMG — APPROVED** [review]
Seems pretty straightforward, and I can confirm in the debugger that students and users are what they're supposed to be. Looks good!

**Comment by thomhickey** [comment]
@mrabbitt I added a catch block, now the test fails more elegantly, if it fails: 

image

**Review by mrabbitt — APPROVED** [review]
LGTM ✅

**Pull request merged**